cmd/podman/search.go

```diff
@@ -172,72 +174,102 @@ func getSearchOutput(term string, registries []string, opts searchOpts, filter s
+			// Check whether query matches filters
+			if !(matchesAutomatedFilter(filter, results[i]) && matchesOfficialFilter(filter, results[i]) && matchesStarFilter(filter, results[i])) {
+				continue
+			}
 		}
-		index := reg
-		arr := strings.Split(reg, ".")
-		if len(arr) > 2 {
-			index = strings.Join(arr[len(arr)-2:], ".")
+		official := ""
+		if results[i].IsOfficial {
+			official = "[OK]"
 		}
-
-		// limit is the number of results to output
-		// if the total number of results is less than the limit, output all
-		// if the limit has been set by the user, output those number of queries
-		limit := maxQueries
-		if len(results) < limit {
-			limit = len(results)
+		automated := ""
+		if results[i].IsAutomated {
+			automated = "[OK]"
 		}
-		if opts.limit != 0 && opts.limit < len(results) {
-			limit = opts.limit
+		description := strings.Replace(results[i].Description, "\n", " ", -1)
+		if len(description) > 44 && !opts.noTrunc {
+			description = description[:descriptionTruncLength] + "..."
 		}
-
-		for i := 0; i < limit; i++ {
-			if len(opts.filter) > 0 {
-				// Check whether query matches filters
-				if !(matchesAutomatedFilter(filter, results[i]) && matchesOfficialFilter(filter, results[i]) && matchesStarFilter(filter, results[i])) {
-					continue
-				}
-			}
-			official := ""
-			if results[i].IsOfficial {
-				official = "[OK]"
-			}
-			automated := ""
-			if results[i].IsAutomated {
-				automated = "[OK]"
-			}
-			description := strings.Replace(results[i].Description, "\n", " ", -1)
-			if len(description) > 44 && !opts.noTrunc {
-				description = description[:descriptionTruncLength] + "..."
-			}
-			name := reg + "/" + results[i].Name
-			if index == "docker.io" && !strings.Contains(results[i].Name, "/") {
-				name = index + "/library/" + results[i].Name
-			}
-			params := searchParams{
-				Index:       index,
-				Name:        name,
-				Description: description,
-				Official:    official,
-				Automated:   automated,
-				Stars:       results[i].StarCount,
-			}
-			paramsArr = append(paramsArr, params)
+		name := registry + "/" + results[i].Name
+		if index == "docker.io" && !strings.Contains(results[i].Name, "/") {
+			name = index + "/library/" + results[i].Name
+		}
+		params := searchParams{
+			Index:       index,
+			Name:        name,
+			Description: description,
+			Official:    official,
+			Automated:   automated,
+			Stars:       results[i].StarCount,
 		}
+		paramsArr = append(paramsArr, params)
 	}
 	return paramsArr, nil
 }
 
 func generateSearchOutput(term string, registries []string, opts searchOpts, filter searchFilterParams) error {
-	searchOutput, err := getSearchOutput(term, registries, opts, filter)
-	if err != nil {
-		return err
+	// searchOutputData is used as a return value for searching in parallel.
+	type searchOutputData struct {
+		data []searchParams
+		err  error
+	}
+
+	// Let's follow Firefox by limiting parallel downloads to 6.
+	sem := semaphore.NewWeighted(6)
```

Comment:
might be good to have '6' define/configurable somewhere in libpod.conf.

Reply:
We have the "magic" number when copying images in parallel, which makes it hard to control via libpod.conf (required some extension to the context).

However, I don't think that is something a user should worry about. 6 seems somehow random but seems to be proven value for nice throughput while playing nice with the servers.